src/js/public/support.js

```diff
@@ -1,95 +1,127 @@
+        validationBy: "onclick",
+        selectors: SELECTORS,
+        messages: MESSAGES,
+        customViewErrors: commonCustomViewErrors,
+        onContainerSuccess: function (container) {
+            currentStep++;
+            supportForm.style.transform = `translateX(-${(currentStep - 1) * 100}%)`;
+            fieldsetsElements[currentStep - 1].classList.add("support-form__fieldset--active");
+            handleProgress(currentStep - 1);
+        },
+    };
 
-    if(!validateProblemDetails()) {
-        e.preventDefault();
+    const configForm = {
+        container: "#support-form",
+        button: "#btn-submit",
+        validationBy: "onclick",
+        messages: {
+            required: MESSAGES.required,
+        },
+        customViewErrors: commonCustomViewErrors,
+        onFormSubmit: function (container) {
+            handleProgress(currentStep);
+            document.querySelector("#support-form").submit();
+        },
+    };
 
-        if (selectCategories.value.trim() === '') {
-            selectCategoriesError.classList.add('support-form__error--active');
-            selectCategoriesError.textContent = 'Campo Obligatorio';
-        }
-        if (problemMessage.value.trim() === '') {
-            problemMessageError.classList.add('support-form__error--active');
-            problemMessageError.textContent = 'Campo Obligatorio';
+    const validatorSteps = new VanillaValidator(configSteps);
+    const validatorForm = new VanillaValidator(configForm);
+
+    function setContainerVisibility(container, isVisible, cls, message) {
+        if (isVisible) {
+        container.classList.add("support-form__error--active", cls);
+        container.innerHTML = message;
+        } else {
+        container.classList.remove("support-form__error--active", cls);
+        container.innerHTML = "";
         }
-    } else {
-        progressName[currentFieldset - 1].classList.add("support-progress__name--active");
-        progressBullet[currentFieldset - 1].classList.add("support-progress__bullet--active");
-        currentFieldset += 1;
     }
-});
+    
+    function updateErrorContainer(field, message, cls) {
+        const containerErrorsID = field.getAttribute("data-errors-id");
+        
+        if (!containerErrorsID) return; 
+    
+        const containerErrors = document.getElementById(containerErrorsID);
+    
+        if (!containerErrors) return; 
+    
+        const shouldDisplayError = Boolean(message);
+    
+        setContainerVisibility(containerErrors, shouldDisplayError, cls, message);
+    }
+
+    const prevButtons = document.querySelectorAll('.prev-step');
+
+    prevButtons.forEach((button) => {
+        button.addEventListener('click', function() {
+            if (currentStep > 1) {
+                currentStep--;
+                supportForm.style.transform = `translateX(-${(currentStep - 1) * 100}%)`;
+                progressName[currentStep - 1].classList.remove('support-progress__name--active');
```

Comment:
```suggestion
                supportForm.style.transform = `translateX(-${(currentStep - 1) * 100}%)`;
                fieldsetsElements.forEach((fieldset) => {
                  fieldset.classList.remove("support-form__fieldset--active");
                });
                fieldsetsElements[currentStep - 1].classList.add("support-form__fieldset--active");
                progressName[currentStep - 1].classList.remove('support-progress__name--active');
```
mismo comentario que arriba

de hecho ya que este codigo es el mismo:

```javascript
fieldsetsElements.forEach((fieldset) => {
  fieldset.classList.remove("support-form__fieldset--active");
});
fieldsetsElements[currentStep - 1].classList.add("support-form__fieldset--active");
```

puedes crear una funcion para esto y nadamas mandarla a llamar